apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/steps/02-step-device-language.tsx

```diff
@@ -0,0 +1,140 @@
+import BigSpinner from "~/renderer/components/BigSpinner";
+import ChangeDeviceLanguagePrompt from "~/renderer/components/ChangeDeviceLanguagePrompt";
+import ChangeDeviceLanguageAction from "~/renderer/components/ChangeDeviceLanguageAction";
+
+type Props = StepProps;
+
+const StepDeviceLanguage = ({
+  updatedDeviceInfo,
+  device,
+  deviceInfo: oldDeviceInfo,
+  transitionTo,
+}: Props) => {
+  const currentLocale = useSelector(languageSelector) as Locale;
+
+  const {
+    availableLanguages: newAvailableLanguages,
+    loaded: newLanguagesLoaded,
+  } = useAvailableLanguagesForDevice(updatedDeviceInfo);
+  const {
+    availableLanguages: oldAvailableLanguages,
+    loaded: oldLanguagesLoaded,
+  } = useAvailableLanguagesForDevice(oldDeviceInfo);
+
+  const [isLanguagePromptOpen, setIsLanguagePromptOpen] = useState<boolean>(false);
+
+  const [languageToInstall, setLanguageToInstall] = useState<Language>("english");
+  const [installingLanguage, setInstallingLanguage] = useState(false);
+
+  const { t } = useTranslation();
+
+  const deviceLocalizationFeatureFlag = useFeature("deviceLocalization");
+
+  const installLanguage = useCallback((language: Language) => {
+    setLanguageToInstall(language);
+    setInstallingLanguage(true);
+  }, []);
+
+  useEffect(() => {
+    if (newLanguagesLoaded && oldLanguagesLoaded) {
+      const deviceLanguageId = updatedDeviceInfo?.languageId;
+      const potentialDeviceLanguage = localeIdToDeviceLanguage[currentLocale];
+
+      const langAvailableForTheFirstTime =
+        potentialDeviceLanguage !== undefined &&
+        !oldAvailableLanguages.includes(potentialDeviceLanguage) &&
+        newAvailableLanguages.includes(potentialDeviceLanguage);
+
+      // firmware version verification is not really needed here, the presence of a language id
+      // indicates that we are in a firmware that supports localization
+      if (
+        langAvailableForTheFirstTime &&
+        deviceLanguageId !== undefined &&
+        idsToLanguage[deviceLanguageId] !== potentialDeviceLanguage &&
+        deviceLocalizationFeatureFlag?.enabled
+      ) {
+        setIsLanguagePromptOpen(true);
+      } else if (
+        oldDeviceInfo?.languageId !== undefined &&
+        oldDeviceInfo?.languageId !== languageIds.english
+      ) {
+        track("Page Manager FwUpdateReinstallLanguage");
+        installLanguage(idsToLanguage[oldDeviceInfo.languageId]);
+      } else {
+        transitionTo("finish");
+      }
+    }
+  }, [
+    newAvailableLanguages,
+    newLanguagesLoaded,
+    oldAvailableLanguages,
+    oldLanguagesLoaded,
+    currentLocale,
+    oldDeviceInfo,
+    updatedDeviceInfo,
+    installLanguage,
+    deviceLocalizationFeatureFlag?.enabled,
+    transitionTo,
+  ]);
+
+  const deviceName = getDeviceModel(device.modelId).productName;
+
+  return (
+    <Flex justifyContent="center" flex={1}>
+      {isLanguagePromptOpen && !installingLanguage && (
```

Comment:
just to be safe in the future, i think it's better to user ternary operators than logical ones inside JSX

Reply:
Fixed